File: Assignment_14/assignment3.c
/////////////////////////////////////////////////////////
//
//  File name :     assignment3.c
//  Description :   Accept number of rows and columns from user and display below pattern.
//  Author :        Aryan S. Pardeshi.
//  Date :          04/06/2025
//
/////////////////////////////////////////////////////////

// Input  :  iRow = 5   iCol = 5
// Output :  a  b  c  d  e
//           1  2  3  4  5
//           a  b  c  d  e
//           1  2  3  4  5
//           a  b  c  d  e

#include <stdio.h>

void Pattern(int iRow, int iCol) 
{
    int i = 0;
    int j = 0;

    for (i = 1; i <= iRow; i++) {
        if (i % 2 != 0) 
        {  
            char ch = 'a';
            for (j = 1; j <= iCol; j++) 
            {
                printf("%c\t", ch);
                ch++;
            }
        } 
        else 
        {
            for (j = 1; j <= iCol; j++) 
            {
                printf("%d\t", j);
            }
        }
        printf("\n");
    }
}

int main() {
    int iValue1 = 0, iValue2 = 0;

    printf("Enter the number of Rows and Columns: ");
    scanf("%d %d", &iValue1, &iValue2);

    Pattern(iValue1, iValue2);

    return 0;
}

File: Assignment_44/assignment3.java
/////////////////////////////////////////////////////////////////////////////////////////////
//
//   File name   :  assignment3.java
//   Description :  Write a program to accept matrix and return largest number from both the diagonals
//   Autor name  :  Aryan Shailendrasingh Pardeshi
//   Date        :  02/08/2025
//
////////////////////////////////////////////////////////////////////////////////////////////

/*
  Input :

  3 2 5 9
  4 3 2 2
  8 4 1 5
  3 9 7 5

  Output : 9
*/

import java.util.*;

class Matrix
{
    private int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int[A][B];
    }

    public void Accept()
    {
        System.out.println("Please Enter the elements of matrix");

        Scanner sobj = new Scanner(System.in);
        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
    }

    public void Display()
    {
        System.out.println("Elements of the matrix are : ");

        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                System.out.print(Arr[i][j]+"\t");
            }
            System.out.println();
        }
    }

    public int MaxDiagonal()
    {
        int i = 0, j = 0, iMax = Arr[0][0];

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                if((i == j) || (i + j == Arr.length - 1))
                {
                    if(Arr[i][j] > iMax)
                    {
                        iMax = Arr[i][j];
                    }
                }
            }
        }

        return iMax;
    }
    
}

class assignment3
{
    public static void main(String A[])
    {
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the number of rows : ");
        int iRow = sobj.nextInt();

        System.out.println("Enter the number of coloums : ");
        int iCol = sobj.nextInt();

        Matrix mobj = new Matrix(iRow, iCol);

        mobj.Accept();

        mobj.Display();
 
        iRet = mobj.MaxDiagonal();
        System.out.println("Maximum number from both the Diagonal is : "+iRet);
    }
}